src/averageAges.js

```diff
@@ -1,61 +1,100 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let menArray = people.filter((man) => {
+    return man.sex === 'm';
+  });
+
+  if (century) {
+    menArray = menArray.filter((man) => {
+      return Math.ceil(man.died / 100) === century;
+    });
+  }
+
+  const ageOfEveryPerson = menArray.map((manAge) => {
+    return manAge.died - manAge.born;
+  });
+
+  const menAverageAge = ageOfEveryPerson.reduce((previousAge, currentAge) => {
+    return previousAge + currentAge;
+  }, 0);
```

Comment:
U can shorten map method. Also u have the same condition in **calculateWomenAverageAge** function. Please move it in separate function
```suggestion
  const menAverageAge = menArray.reduce((acc, person) => {
    return previousAge + (person.diead - person.born);
  }, 0);
```

![image](https://github.com/mate-academy/js_average-ages/assets/54936044/32a8d79d-3953-464b-8084-e2da94187b9f)